pulp_container/tests/functional/api/test_pull_content.py

```diff
@@ -186,117 +100,100 @@ def test_create_empty_blob_on_the_fly(self):
-
-        local_url = (
-            urljoin(self.cfg.get_base_url(), self.distribution_with_repo.base_path)
-            + PULP_HELLO_WORLD_LINUX_TAG
+        local_registry.pull(self.distribution_with_repo.base_path + PULP_HELLO_WORLD_LINUX_TAG)
+        local_image = local_registry.inspect(
+            self.distribution_with_repo.base_path + PULP_HELLO_WORLD_LINUX_TAG
         )
 
-        registry.pull(local_url)
-        self.teardown_cleanups.append((registry.rmi, local_url))
-        local_image = registry.inspect(local_url)
-
-        registry.pull(REGISTRY_V2_REPO_HELLO_WORLD + PULP_HELLO_WORLD_LINUX_TAG)
-        self.teardown_cleanups.append(
-            (registry.rmi, REGISTRY_V2_REPO_HELLO_WORLD + PULP_HELLO_WORLD_LINUX_TAG)
+        registry_client.pull(REGISTRY_V2_REPO_HELLO_WORLD + PULP_HELLO_WORLD_LINUX_TAG)
+        remote_image = registry_client.inspect(
+            REGISTRY_V2_REPO_HELLO_WORLD + PULP_HELLO_WORLD_LINUX_TAG
         )
-        remote_image = registry.inspect(REGISTRY_V2_REPO_HELLO_WORLD + PULP_HELLO_WORLD_LINUX_TAG)
 
-        self.assertEqual(local_image[0]["Id"], remote_image[0]["Id"])
+        registry_client.rmi(REGISTRY_V2_REPO_HELLO_WORLD + PULP_HELLO_WORLD_LINUX_TAG)
+        assert local_image[0]["Id"] == remote_image[0]["Id"]
 
-    def test_pull_nonexistent_image(self):
+    def test_pull_nonexistent_image(self, local_registry):
         """Verify that a client cannot pull nonexistent image from Pulp.
 
         1. Using the RegistryClient try to pull nonexistent image from Pulp.
         2. Assert that error is occurred and nothing has been pulled.
         """
-        registry = cli.RegistryClient(self.cfg)
-        registry.raise_if_unsupported(unittest.SkipTest, "Test requires podman/docker")
-        registry.login("-u", "admin", "-p", "password", self.registry_name)
+        with pytest.raises(subprocess.CalledProcessError):
+            local_registry.pull("inexistentimagename")
 
-        local_url = urljoin(self.cfg.get_base_url(), "inexistentimagename")
-        with self.assertRaises(exceptions.CalledProcessError):
-            registry.pull(local_url)
-
-    def test_pull_nonexistent_blob(self):
+    def test_pull_nonexistent_blob(self, bindings_cfg):
         """
         Verify that a GET request to a nonexistent BLOB will be properly handled
         instead of outputting a stacktrace.
         """
         blob_path = "/v2/{}/blobs/{}".format(self.distribution_with_repo.base_path, EMPTY_JSON)
-        non_existing_blob_url = urljoin(self.cfg.get_base_url(), blob_path)
+        non_existing_blob_url = urljoin(bindings_cfg.host, blob_path)
 
         auth = get_auth_for_url(non_existing_blob_url)
         content_response = requests.get(non_existing_blob_url, auth=auth)
-        assert content_response.status_code, 404
+        assert content_response.status_code == 404
 
 
-class PullOnDemandContentTestCase(unittest.TestCase):
+class TestPullOnDemandContent:
```

Comment:
Is it not possible for `TestPullContent` and `TestPullOnDemandContent` to share some of this code instead of duplicating so much of it?  It looks like much of it was copied and pasted at one point.

Reply:
This shouldn't be interpreted as a blocker to the immediate goal of dropping pulp-smash ^